plugins/frontend/backstage-plugin-github-insights/src/components/Widgets/MarkdownContent/MarkdownContent.tsx

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2021 Larder Software Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useCallback, useEffect, useState } from 'react';
+import { Alert } from '@material-ui/lab';
+import {
+  Progress,
+  MarkdownContent as RawMarkdownContent,
+} from '@backstage/core-components';
+import {
+  useApi,
+  githubAuthApiRef,
+  SessionState,
+  useApiHolder,
+  ApiHolder,
+  errorApiRef,
+  ErrorApi,
+} from '@backstage/core-plugin-api';
+import { MarkdownContentProps } from './types';
+import { Button, Grid, Typography, Tooltip } from '@material-ui/core';
+import useAsync from 'react-use/lib/useAsync';
+import { githubApiRef, GithubClient, GithubApi } from '../../../apis';
+
+const getGithubClient = (apiHolder: ApiHolder, errorApi: ErrorApi) => {
+  let githubClient: GithubApi | undefined = apiHolder.get(githubApiRef);
+  if (!githubClient) {
+    const auth = apiHolder.get(githubAuthApiRef);
+    if (auth) {
+      githubClient = new GithubClient({ githubAuthApi: auth, errorApi });
+    }
+  }
+  if (!githubClient) {
+    throw new Error(
+      'The MarkdownCard component Failed to get the github client',
+    );
+  }
+  return githubClient;
+};
+
+const GithubFileContent = (props: MarkdownContentProps) => {
+  const { preserveHtmlComments } = props;
+  const apiHolder = useApiHolder();
+  const errorApi = useApi(errorApiRef);
+
+  const { value, loading, error } = useAsync(async () => {
+    const githubClient = getGithubClient(apiHolder, errorApi);
+    return githubClient.getContent({ ...props });
+  }, [apiHolder]);
+
+  const transformImageUri = useCallback(
+    (href: string) => {
+      return value?.media[href] || href;
```

Comment:
This function replaces the media links compiled by the GithubClient#getContent function.